website_sale_attribute_filter_price/controllers/website_sale.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2020 Tecnativa - Alexandre Díaz
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import http
+from odoo.http import request
+
+from odoo.addons.website_sale.controllers.main import WebsiteSale
+
+
+class WebsiteSale(WebsiteSale):
+    def _get_search_domain(self, search, category, attrib_values):
+        domain = super()._get_search_domain(search, category, attrib_values)
+        price_vals = request.env.context.get("price_vals")
+        if price_vals:
+            domain += [
+                "&",
+                ("list_price", ">=", price_vals[0]),
+                ("list_price", "<=", price_vals[1]),
+            ]
+        return domain
+
+    @http.route()
+    def shop(self, page=0, category=None, search="", ppg=False, **post):
+        # Get maximium price
+        _pricelist_context, pricelist = self._get_pricelist_context()
+        request.context = dict(
+            request.context,
+            pricelist=pricelist.id,
+            partner=request.env.user.partner_id)
+        product_id = request.env['product.template'].with_context(
+            prefetch_fields=False
+        ).search(
+            self._get_search_domain(False, False, False),
+            order='list_price DESC', limit=1)
+        max_price = product_id.list_price
+        # Obtain and sanitize user values
+        try:
+            custom_min_price = float(post.get("min_price"))
+        except (ValueError, TypeError):
+            custom_min_price = 0.0
+        try:
+            custom_max_price = float(post.get("max_price"))
+        except (ValueError, TypeError):
+            custom_max_price = max_price
+        if custom_min_price > custom_max_price:
+            custom_max_price, custom_max_price = custom_min_price, custom_max_price
+        # Call Super with context
+        request.env.context = dict(request.env.context,
+            price_vals = [custom_min_price, custom_max_price]
+        )
```

Comment:
The lint error comes from here
```suggestion
        request.env.context = dict(
            request.env.context,
            price_vals=[custom_min_price, custom_max_price])
```